tap_klaviyo/__init__.py

```diff
@@ -87,13 +109,30 @@ def load_schema(name):
     return json.load(open(get_abs_path('schemas/{}.json'.format(name))))
 
 
+def build_metadata(name, key_properties):
+    schema = load_schema(name)
+
+    mdata = metadata.new()
+    mdata = metadata.write(mdata, (), 'table-key-properties', key_properties)
+
+    for field in schema["properties"].keys():
+        mdata = metadata.write(mdata, ('properties', field), 'inclusion', 'available')
+
+    return metadata.to_list(mdata)
+
+def stream_is_selected(mdata):
+    return mdata.get((), {}).get('selected', False)
```

Comment:
what is the first argument `()` to `mdata.get`?

a comment explaining that, or even better a comment showing the shape (or example) of `mdata, would be good.

Reply:
This is copied from [this](https://github.com/hotgluexyz/tap-klaviyo/blob/master/tap_klaviyo/__init__.py#L120) line of code and I have no idea what it means but it's keying into a structure that looks like:
```
"metadata": [
        {
          "breadcrumb": [],
          "metadata": {
            "table-key-properties": "uuid",
            "selected": true
          }
        }
]
```

Reply:
After some investigating, it's the Singer's python library's way of basically specifying key-paths into a dict. `()` appears to be equivalent to keying into the top-level of fields in the dictionary. See "breadcrumb" [here](https://github.com/singer-io/singer-python/blob/master/singer/metadata.py#L22).